src/store/modules/comments/actions/index.js

```diff
@@ -0,0 +1,111 @@
+import axios from 'axios';
+import * as Toastr from 'toastr';
+import { getSuccessResponse, getErrorResponse } from '@utils/getResponse';
+import getUserInfo from '../../../../utils/getUserFromToken';
+import {
+  GET_COMMENTS_PENDING,
+  GET_COMMENTS_SUCCESS,
+  GET_COMMENTS_FAILURE,
+  POST_COMMENT_PENDING,
+  POST_COMMENT_SUCCESS,
+  POST_COMMENT_FAILURE
+} from '../actionTypes';
+
+const removeQuotesFromString = str => str.trim().slice(1, str.length - 1);
+
+export const getCommentPending = () => ({
+  type: GET_COMMENTS_PENDING,
+  payload: {
+    status: 'commentPending',
+    error: null,
+    comments: []
+  }
+});
+
+export const getCommentSuccess = comments => ({
+  type: GET_COMMENTS_SUCCESS,
+  payload: {
+    status: 'commentSuccess',
+    error: null,
+    comments
+  }
+});
+
+export const getCommentFailure = () => ({
+  type: GET_COMMENTS_FAILURE,
+  payload: {
+    status: 'commentFailure',
+    comments: []
+  }
+});
+
+export const postCommentPending = () => ({
+  type: POST_COMMENT_PENDING,
+  payload: {
+    status: 'postcommentPending',
+    error: null
+  }
+});
+
+export const postCommentSuccess = comment => ({
+  type: POST_COMMENT_SUCCESS,
+  payload: {
+    status: 'postCommentSuccess',
+    error: null,
+    comment
+  }
+});
+
+export const postCommentFailure = error => ({
+  type: POST_COMMENT_FAILURE,
+  payload: {
+    status: 'postcommentFailure',
+    error
+  }
+});
+export const getComments = articleId => async dispatch => {
+  const token = localStorage.getItem('token');
```

Comment:
Abstract this functionality to a separate utility.

Checking localstorage is i/o which slow. It's synchronous, meaning it blocks the main thread and prevents you application from doing anything else or responding to user input.

Localstorage operations should also be wrapped in a try/catch block.

Reply:
Done! Thank you